### PR TITLE
fix(developer): validate kmconvert parameters

### LINKED ISSUE #5472: bug(developer): kmconvert should check input parameters
**Describe the bug**

`kmconvert` happily accepts invalid ids and languages that `kmpcomp` then refuses to compile.

**To Reproduce**

Run the following commands:

```
kmconvert template -id FooBar -name "Foo Bar" -languages en-us
cd FooBar
kmcomp FooBar.kpj
```

**Expected behavior**

`kmconvert` refuses invalid id `FooBar` and non-canonical languages.

**Actual behavior**

```
D:\tmp\FooBar>kmcomp FooBar.kpj
Keyman Developer Compiler (32-bit)
Version 14.0.236.0, Copyright c SIL International.  All Rights Reserved.

FooBar.kmn: Warning: 2000 The keyboard file FooBar.kmn does not follow the recommended filename conventions. The name should be all lower case, include only alphanumeric characters and underscore (_), and should not start with a digit.
FooBar.kmn: Warning: 2000 The keyboard file FooBar.kvks does not follow the recommended filename conventions. The name should be all lower case, include only alphanumeric characters and underscore (_), and should not start with a digit.
FooBar.kmn: Compiling 'D:\tmp\FooBar\source\FooBar.kmn' for macOS, Desktop devices...
FooBar.kmn: Failure: 'D:\tmp\FooBar\source\FooBar.kmn' was not compiled successfully for macOS, Desktop devices.
```

After fixing the id to `foo_bar` and trying to compile the output is:

```
D:\tmp\foo_bar>kmcomp foo_bar.kpj
Keyman Developer Compiler (32-bit)
Version 14.0.236.0, Copyright c SIL International.  All Rights Reserved.

foo_bar.kmn: Compiling 'D:\tmp\foo_bar\source\foo_bar.kmn' for macOS, Desktop devices...
foo_bar.kmn: Success: 'D:\tmp\foo_bar\source\foo_bar.kmn' was compiled successfully for macOS, Desktop devices to 'D:\tmp\foo_bar\build\foo_bar.kmx'.
foo_bar.kmn: Compiling 'D:\tmp\foo_bar\source\foo_bar.kmn' for iPhone, Android tablet, Mobile devices, Tablet devices...
foo_bar.kmn: Success: 'D:\tmp\foo_bar\source\foo_bar.kmn' was compiled successfully for iPhone, Android tablet, Mobile devices, Tablet devices to 'D:\tmp\foo_bar\build\foo_bar.js'.
foo_bar.kps: Compiling package foo_bar.kps...
foo_bar.kps: Warning: Keyboard with ID 'foo_bar' has a BCP 47 warning: 'en-US' is a valid tag but is not canonical: it should be 'en'.
foo_bar.kps: Failure: 'D:\tmp\foo_bar\source\foo_bar.kps' was not compiled successfully.
```

---

**Keyman for Windows:**
 - OS: Windows 10
 - Keyman version: 14.0.278

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #5472.

Verifies most command line parameters for kmconvert. Note that non-canonical BCP-47 tags emit a warning rather than an error.

Noticed also that unit tests were not actually stopping the build if they failed...

@keymanapp-test-bot skip

### CODE REVIEW

File: windows/src/test/unit-tests/kmconvert/Keyman.Developer.System.Test.KMConvertParametersTest.pas [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+procedure TKMConvertParametersTest.TestModelIdAuthor;
+var
+  k: TKMConvertParameters;
+begin
+  k.OnOutputText := procedure(s: string) begin Log(TLogLevel.Information, s); end;
+  Assert.IsTrue(k.CheckParams(['template', '-id-author', 'myself']));
+  Assert.IsTrue(k.CheckParams(['template', '-id-author', 'my_self']));
+  Assert.IsFalse(k.CheckParams(['template', '-id-author', 'my-self']));
+  Assert.IsFalse(k.CheckParams(['template', '-id-author', 'Myself']));
+  Assert.IsFalse(k.CheckParams(['template', '-id-author', '$myself$']));
```
Comment (darcywong00) [inline windows/src/test/unit-tests/kmconvert/Keyman.Developer.System.Test.KMConvertParametersTest.pas:146]: Are there restrictions on valid `-id-author values`?

https://help.keyman.com/developer/current-version/context/kmconvert#toc-lexical-model-parameters

Comment (mcdurdin) [reply windows/src/test/unit-tests/kmconvert/Keyman.Developer.System.Test.KMConvertParametersTest.pas:146]: Yes, the -id-author and -id-uniq values must be valid tokens, i.e. lower case a-z, 0-9, and underscore. Should start with alpha or underscore. The `author.bcp47.uniq` value has the following regex: 

```
^[a-z_][a-z0-9_]*\.[a-z_][a-z0-9_-]*\.[a-z_][a-z0-9_]*$
```

Comment (darcywong00) [reply windows/src/test/unit-tests/kmconvert/Keyman.Developer.System.Test.KMConvertParametersTest.pas:146]: How about a test?
```
Assert.IsTrue(k.checkParams(['template', 'id-author', '_myself']));
```

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.227-beta](https://keyman.com/downloads/releases/beta/15.0.227)

File: windows/src/test/unit-tests/kmconvert/Keyman.Developer.System.Test.KMConvertParametersTest.pas (hunk 1 above)
Comment (mcdurdin) [reply windows/src/test/unit-tests/kmconvert/Keyman.Developer.System.Test.KMConvertParametersTest.pas:146]: Ah, missed this comment before I merged! Next time I guess?

**Pull request merged**